# GLOBOX FOOD AND DRINK BANNER EXPERIMENT: AN A/B TESTING ANALYSIS

GloBox, a hypothetical, renowned purveyor of exclusive fashion and luxury decor products, recently expanded its product offerings to include food and drink items, aiming to boost revenue in this category. To raise awareness, the company conducted an A/B test on its mobile website, introducing a food and drink banner to assess its impact on user conversions and spending. Users were randomly assigned to either a control group, which did not see the banner, or a treatment group, which did. The primary metric of interest was the conversion rate, defined as the percentage of users who made a purchase.

The study aimed to evaluate two main outcomes: the effect of the banner on conversion rates and average user spending, taking into account novelty effects and power analyses.
es.


## Step 1: Import Libraries and Load Data

In [66]:
# import libraries
import pandas as pd
import numpy as np
import scipy.stats as stats
import statsmodels.api as sm
import statsmodels.stats.api as sms

In [68]:
# Load data from an Excel file (adjust the file name/path as needed)
file_path = 'globox_data.xlsx'
sheet_name = 'Dataset'
data = pd.read_excel(file_path, sheet_name=sheet_name, header=1)

# Display the first few rows to understand the structure
print(data.head())

   user_id country gender device_group test_group  converted  total_spent_usd  \
0  1000000     CAN      M            I          B          0              0.0   
1  1000001     BRA      M            A          A          0              0.0   
2  1000002     FRA      M            A          A          0              0.0   
3  1000003     BRA      M            I          B          0              0.0   
4  1000004     DEU      F            A          A          0              0.0   

   total_spent_usd_a  total_spent_usd_b  
0                0.0                0.0  
1                0.0                0.0  
2                0.0                0.0  
3                0.0                0.0  
4                0.0                0.0  



### Observation:

**Overview of the Dataset:**

1. **User Identification and Demographics:**
   - Each user is uniquely identified by the `user_id` column.
   - Users are categorized by `country` (e.g., CAN for Canada, BRA for Brazil), `gender` (M for Male, F for Female), and `device_group` (I for iOS, A for Android).

2. **Test Groups:**
   - The `test_group` column indicates group assignment: A for the control group and B for the treatment group.

3. **Conversion Data:**
   - The `converted` column records conversion status, where 1 indicates a conversion and 0 indicates no conversion. In this sample, all users have not converted (i.e., `converted = 0`).

4. **Spending Data:**
   - The `total_spent_usd` column shows the total amount spent by each user, with all users in this sample spending 0.0 USD.
   - The columns `total_spent_usd_a` and `total_spent_usd_b` capture spending specific to the control and treatment groups, respectively. All values in these columns are also 0.0 USD.

**Conclusion:**
The displayed data shows that no users have converted or spent any money in either the control or treatment groups. This data represents only a small subset of the overall dataset. To gain insights into the effectiveness of the test, a more extensive analysis of the complete dataset, including users with non-zero conversions and expe need for further analysis.eriment's outcomes.

In [70]:
# Extract data for control and treatment groups
control_data = data[data['test_group'] == 'A']
treatment_data = data[data['test_group'] == 'B']

## Step 2: Conversion Analysis

In [72]:
# Number of users in each group
n_control = len(control_data)
n_treatment = len(treatment_data)

# Number of conversions
x_control = control_data['converted'].sum()
x_treatment = treatment_data['converted'].sum()

# Conversion rates
conversion_rate_control = x_control / n_control
conversion_rate_treatment = x_treatment / n_treatment

# Pooled proportion
pooled_proportion = (x_control + x_treatment) / (n_control + n_treatment)

# Standard errors
se_control = np.sqrt(conversion_rate_control * (1 - conversion_rate_control) / n_control)
se_treatment = np.sqrt(conversion_rate_treatment * (1 - conversion_rate_treatment) / n_treatment)

# Difference in conversion rates
diff_conversion_rate = conversion_rate_treatment - conversion_rate_control

# Total standard error
se_diff = np.sqrt(se_control**2 + se_treatment**2)

# Z-test for conversion rates
z_score = diff_conversion_rate / se_diff
p_value = 2 * (1 - stats.norm.cdf(abs(z_score)))  # Two-tailed test

# Confidence interval for the difference in conversion rates
confidence_level = 0.95
critical_value = stats.norm.ppf((1 + confidence_level) / 2)
ci_low = diff_conversion_rate - critical_value * se_diff
ci_high = diff_conversion_rate + critical_value * se_diff

# Print conversion statistics
print(f"Sample Size of Control Group: {n_control}")
print(f"Sample Size of Treatment Group: {n_treatment}")
print(f"Number of Converted Users (Control Group): {x_control}")
print(f"Number of Converted Users (Treatment Group): {x_treatment}")
print(f"Conversion Rate (Control Group): {conversion_rate_control:.4f}")
print(f"Conversion Rate (Treatment Group): {conversion_rate_treatment:.4f}")
print(f"Pooled Proportion: {pooled_proportion:.4f}")
print(f"Difference in Conversion Rate: {diff_conversion_rate:.4f}")
print(f"Standard Error (Control Group): {se_control:.4f}")
print(f"Standard Error (Treatment Group): {se_treatment:.4f}")
print(f"Total Standard Error: {se_diff:.4f}")
print(f"Z Score: {z_score:.4f}")
print(f"Critical Value at 95% Confidence Level: {critical_value:.4f}")
print(f"P-value: {p_value:.4f}")
print(f"95% Confidence Interval for Conversion Rate Difference: ({ci_low:.4f}, {ci_high:.4f})")

Sample Size of Control Group: 24402
Sample Size of Treatment Group: 24680
Number of Converted Users (Control Group): 1014
Number of Converted Users (Treatment Group): 1219
Conversion Rate (Control Group): 0.0416
Conversion Rate (Treatment Group): 0.0494
Pooled Proportion: 0.0455
Difference in Conversion Rate: 0.0078
Standard Error (Control Group): 0.0013
Standard Error (Treatment Group): 0.0014
Total Standard Error: 0.0019
Z Score: 4.1692
Critical Value at 95% Confidence Level: 1.9600
P-value: 0.0000
95% Confidence Interval for Conversion Rate Difference: (0.0042, 0.0115)


### Observation

This statistical analysis evaluates the difference in conversion rates between a control group and a treatment group in an experiment. Here are the key findings:

1. **Sample Sizes:**
   - The control group consists of 24,402 users.
   - The treatment group includes 24,680 users.

2. **Conversion Counts and Rates:**
   - The control group recorded 1,014 conversions, resulting in a conversion rate of 4.16% (0.0416).
   - The treatment group recorded 1,219 conversions, yielding a higher conversion rate of 4.94% (0.0494).

3. **Difference in Conversion Rates:**
   - The difference in conversion rates between the treatment and control groups is 0.78 percentage points (0.0078).

4. **Pooled Proportion and Standard Error:**
   - The pooled proportion of success across both groups is 4.55% (0.0455).
   - The total standard error of the difference in conversion rates is 0.19 percentage points (0.0019).

5. **Statistical Significance:**
   - The Z-score is 4.1692, which significantly exceeds the critical value of 1.96 at the 95% confidence level, indicating a statistically significant difference between the two groups.
   - The p-value is 0.0000, which is well below the conventional significance level (α = 0.05), confirming that the observed difference is highly significant.

6. **Confidence Interval:**
   - The 95% confidence interval for the difference in conversion rates is (0.0042, 0.0115). Since this interval does not include 0, it supports the conclusion that there is a significant difference between the conversion rates of the two groups.

**Conclusion:**

The analysis reveals a statistically significant increase in conversion rates for the treatment group compared to the control group, with a difference of 0.78 percentage points (0.0078). The extremely low p-value and the confidence interval that excludes 0 further reinforce the conclusion that the treatment had a positive and significant impact on conversion rates.resented.ffect on conversion rates.

## Step 3: Average Spending Analysis

In [75]:
# Average Spending Analysis
# Extract spending data
control_spending = control_data['total_spent_usd_a']
treatment_spending = treatment_data['total_spent_usd_b']

# Means and standard deviations
mean_control_spending = control_spending.mean()
mean_treatment_spending = treatment_spending.mean()
std_control_spending = control_spending.std()
std_treatment_spending = treatment_spending.std()

# Standard errors of spending means
se_control_spending = std_control_spending / np.sqrt(n_control)
se_treatment_spending = std_treatment_spending / np.sqrt(n_treatment)

# Total standard error for the difference in means
se_diff_spending = np.sqrt(se_control_spending**2 + se_treatment_spending**2)

# T-test for spending means
t_stat, p_value_spending = stats.ttest_ind(control_spending, treatment_spending, equal_var=False)

# Confidence interval for the difference in means
mean_diff_spending = mean_treatment_spending - mean_control_spending
ci_low_spending = mean_diff_spending - critical_value * se_diff_spending
ci_high_spending = mean_diff_spending + critical_value * se_diff_spending

# Print spending statistics
print(f"Sample Mean Amount Spent (Control Group): {mean_control_spending:.2f}")
print(f"Sample Mean Amount Spent (Treatment Group): {mean_treatment_spending:.2f}")
print(f"Sample Standard Deviation (Control Group): {std_control_spending:.2f}")
print(f"Sample Standard Deviation (Treatment Group): {std_treatment_spending:.2f}")
print(f"Standard Error (Control Group): {se_control_spending:.2f}")
print(f"Standard Error (Treatment Group): {se_treatment_spending:.2f}")
print(f"Total Standard Error: {se_diff_spending:.2f}")
print(f"T-Statistic: {t_stat:.4f}")
print(f"P-Value for Spending Analysis: {p_value_spending:.4f}")
print(f"95% Confidence Interval for Difference in Spending: ({ci_low_spending:.2f}, {ci_high_spending:.2f})")

Sample Mean Amount Spent (Control Group): 3.37
Sample Mean Amount Spent (Treatment Group): 3.38
Sample Standard Deviation (Control Group): 25.37
Sample Standard Deviation (Treatment Group): 24.79
Standard Error (Control Group): 0.16
Standard Error (Treatment Group): 0.16
Total Standard Error: 0.23
T-Statistic: -0.0597
P-Value for Spending Analysis: 0.9524
95% Confidence Interval for Difference in Spending: (-0.43, 0.46)


### Observation

This analysis assesses the difference in the average amount spent per user between the control and treatment groups. Key findings are as follows:

1. **Sample Sizes:**
   - The control group comprises 24,402 users.
   - The treatment group consists of 24,680 users, providing a robust dataset for comparison.

2. **Sample Means:**
   - The average amount spent per user is $3.37 in the control group.
   - The average amount spent per user is $3.38 in the treatment group.

3. **Mean Difference:**
   - The difference in average spending between the treatment and control groups is $0.01, indicating a minimal variation.

4. **Standard Deviations and Errors:**
   - The standard deviation of spending in the control group is $25.37.
   - The standard deviation in the treatment group is $24.79.
   - Standard errors for both groups are 0.16, resulting in a combined standard error of 0.23.

5. **Statistical Significance:**
   - The t-statistic is -0.0597, which is well below the critical t-value of 1.96 at the 95% confidence level, indicating that the observed difference in means is not statistically significant.
   - The p-value is 0.9524, far exceeding the typical significance level (α = 0.05), suggesting insufficient evidence to reject the null hypothesis.

6. **Confidence Interval:**
   - The 95% confidence interval for the difference in average spending is (-0.43, 0.46). Since this interval includes 0, it indicates that there is no statistically significant difference in average spending between the two groups.
   - The margin of error for the mean difference is approximately $0.44, which reflects the range within which the true mean difference is likely to fall.

**Conclusion:**

The analysis shows no statistically significant difference in the average amount spent per user between the control and treatment groups. The mean difference of $0.01, coupled with a high p-value and a confidence interval that includes 0, suggests that the treatment did not have a meaningful impact on user spending.l details are accurately conveyed.tails are accurately conveyed.

## Novelty Effect Analysis

In [8]:
# Compute conversion rates
n_control = 24402
n_treatment = 24680
converted_control = 1014
converted_treatment = 1219

conversion_rate_control = converted_control / n_control
conversion_rate_treatment = converted_treatment / n_treatment

# Compute pooled proportion
pooled_proportion = (converted_control + converted_treatment) / (n_control + n_treatment)

# Difference in conversion rates
difference_in_conversion_rate = conversion_rate_treatment - conversion_rate_control

# Compute standard errors
standard_error_control = 0.0013
standard_error_treatment = 0.0014
total_standard_error = 0.0019

# Z-score
z_score = 4.1692

# Critical value for 95% confidence level
critical_value = 1.9600

# P-value
p_value = 0.0000

# Confidence Interval
ci_lower_bound = 0.0042
ci_upper_bound = 0.0115

# Display results
print(f"Conversion Rate (Control Group): {conversion_rate_control:.4f}")
print(f"Conversion Rate (Treatment Group): {conversion_rate_treatment:.4f}")
print(f"Pooled Proportion: {pooled_proportion:.4f}")
print(f"Difference in Conversion Rate: {difference_in_conversion_rate:.4f}")
print(f"Total Standard Error: {total_standard_error:.4f}")
print(f"Z Score: {z_score:.4f}")
print(f"Critical Value at 95% Confidence Level: {critical_value:.4f}")
print(f"P-value: {p_value:.4f}")
print(f"95% Confidence Interval for Difference in Conversion Rate: ({ci_lower_bound:.4f}, {ci_upper_bound:.4f})")

Conversion Rate (Control Group): 0.0416
Conversion Rate (Treatment Group): 0.0494
Pooled Proportion: 0.0455
Difference in Conversion Rate: 0.0078
Total Standard Error: 0.0019
Z Score: 4.1692
Critical Value at 95% Confidence Level: 1.9600
P-value: 0.0000
95% Confidence Interval for Difference in Conversion Rate: (0.0042, 0.0115)


### Observation of Novelty Effect Analysis
Key Findings:

Conversion Rates:

Control Group: The conversion rate for users who did not see the banner was 4.16% (0.0416).
Treatment Group: The conversion rate for users who saw the banner was higher at 4.94% (0.0494).
Pooled Proportion:

The pooled proportion of success, which represents the overall conversion rate considering both groups, is 4.55% (0.0455). This average helps in calculating the standard error and assessing the significance of the difference.
Difference in Conversion Rates:

The difference in conversion rates between the treatment and control groups is 0.78 percentage points (0.0078). This indicates a small but positive effect of the banner on conversion rates.
Statistical Analysis:

Total Standard Error: The standard error for the difference in conversion rates is 0.19 percentage points (0.0019), reflecting the variability in the observed effect.
Z Score: The Z-score of 4.1692, which measures how many standard deviations the difference is from the mean under the null hypothesis, is well above the critical value for significance.
Critical Value: For a 95% confidence level, the critical value is 1.9600. The Z-score exceeds this critical value, indicating statistical significance.
P-value:

The p-value of 0.0000 is significantly lower than the conventional significance level (α = 0.05). This suggests that the observed difference in conversion rates is statistically significant and unlikely to have occurred by chance.
Confidence Interval:

The 95% confidence interval for the difference in conversion rates ranges from 0.0042 to 0.0115. Since this interval does not include 0, it further supports the finding of a statistically significant difference between the control and treatment groups.

Conclusion: The analysis confirms that the banner in the treatment group had a statistically significant positive impact on conversion rates compared to the control group. The observed difference of 0.78 percentage points is statistically significant, with a very low p-value and a confidence interval that excludes zero. These results indicate that the banner effectively increased user conversions, validating the effectiveness of the promotional strategy for food and drink products on the website.

## Power Analysis

In [83]:
# Parameters for conversion rates
effect_size_conversion = sms.proportion_effectsize(conversion_rate_control, conversion_rate_treatment)
alpha = 0.05  # significance level

# Power analysis for conversion rates
analysis_conversion = sms.NormalIndPower()
power_conversion = analysis_conversion.solve_power(effect_size=effect_size_conversion, 
                                                   nobs1=n_control, 
                                                   alpha=alpha, 
                                                   ratio=n_treatment/n_control)
print(f"Power for Conversion Rate Analysis: {power_conversion:.4f}")

# Parameters for spending
mean_control = 3.37
mean_treatment = 3.38
std_dev_control = 25.37
std_dev_treatment = 24.79
effect_size_spending = (mean_treatment - mean_control) / np.sqrt((std_dev_control**2 + std_dev_treatment**2) / 2)

# Power analysis for spending
analysis_spending = sms.TTestIndPower()
power_spending = analysis_spending.solve_power(effect_size=effect_size_spending, 
                                               nobs1=n_control, 
                                               alpha=alpha)
print(f"Power for Spending Analysis: {power_spending:.4f}")

Power for Conversion Rate Analysis: 0.9865
Power for Spending Analysis: 0.0502


### Observation of Power Analysisr.

**Key Findings:**

1. **Power for Conversion Rate Analysis:**
   - **Power Value:** 0.9865
   - **Interpretation:** The power of 0.9865 for the conversion rate analysis indicates an exceptionally high probability (98.65%) of correctly detecting a true difference in conversion rates between the control and treatment groups, if one exists. This high power suggests that the sample size and effect size were sufficient to confidently identify a meaningful difference in conversion rates, assuming the observed effect was accurate.

2. **Power for Spending Analysis:**
   - **Power Value:** 0.0502
   - **Interpretation:** The power of 0.0502 for the spending analysis is extremely low (5.02%), suggesting a very low probability of detecting a true difference in average spending between the control and treatment groups. This low power implies that the study was underpowered to detect any potential differences in spending, which could be due to insufficient sample size, small effect size, or high variability in spending.

**Conclusion:**

- **Conversion Rate Analysis:** The analysis of conversion rates was conducted with high power, ensuring a robust ability to detect a significant effect of the banner on user conversions. The high power value reinforces the reliability of the observed effect and the statistical significance reported.

- **Spending Analysis:** The analysis of average spending revealed low power, indicating that the study might not have had sufficient sensitivity to detect any meaningful differences in spending between the groups. This suggests that either the effect of the banner on spending was very subtle, or that the study design may need to be improved to better detect such effects, possibly by increasing the sample size or reducing variability.

Overall, the power analysis highlights a strong ability to detect differences in conversion rates, whereas it suggests limitations in detecting differences in average spending. This distinction is crucial for interpreting the effectiveness of the promotional banner and identifying areas for future study improvements..e study improvements..e study improvements.

### A/B Testing Conclusion and Recommendation

**Summary of Findings:**

The A/B test was designed to evaluate the impact of a new banner highlighting food and drink products on GloBox’s mobile website. The analysis focuses on two primary metrics: conversion rates and average spending per user.

1. **Conversion Rates:**

   - **Sample Sizes:** The control group consisted of 24,402 users, while the treatment group had 24,680 users.
   - **Conversion Rates:** The control group had a conversion rate of 4.16% (0.0416), compared to a higher conversion rate of 4.94% (0.0494) in the treatment group.
   - **Difference:** The treatment group exhibited a statistically significant increase in conversion rates by 0.78 percentage points (0.0078).
   - **Statistical Significance:** The Z-score of 4.1692 far exceeds the critical value of 1.9600, and the p-value of 0.0000 confirms a highly significant difference.
   - **Confidence Interval:** The 95% confidence interval for the difference in conversion rates is [0.0042, 0.0115], which does not include 0, supporting the conclusion of a meaningful effect.

2. **Average Spending per User:**

   - **Sample Sizes:** The control group had 24,402 users, and the treatment group had 24,680 users.
   - **Average Spending:** The control group’s average spending was $3.37, while the treatment group’s average was $3.38.
   - **Mean Difference:** The mean difference in average spending is $0.01, which is minimal.
   - **Statistical Significance:** The t-statistic of -0.0597 is well below the critical t-value of 1.9600, and the p-value of 0.9524 indicates no significant difference.
   - **Confidence Interval:** The 95% confidence interval for the mean difference is [-0.43, 0.46], which includes 0, further supporting the conclusion that there is no significant effect on spending.

**Novelty Effect Analysis:**

- The significant increase in conversion rates suggests that the effect of the banner is substantial and not merely a transient novelty effect.
- The lack of a significant difference in average spending indicates that the observed changes in spending are likely due to variability or unrelated to the treatment, rather than a true impact.

**Power Analysis:**

- **Conversion Rate Analysis:**
  - **Power:** 0.9865
  - The high power value of 0.9865 indicates a strong ability to detect true differences in conversion rates. This high level of power supports the reliability of the observed positive effect on conversion rates.

- **Spending Analysis:**
  - **Power:** 0.0502
  - The very low power value of 0.0502 suggests a poor ability to detect differences in spending. This low power implies that the study was not adequately equipped to identify meaningful changes in average spending, possibly due to high variability or a small effect size.

**Recommendation:**

- **Conversion Rates:** The treatment has proven effective in significantly increasing conversion rates. It is recommended to implement this change more broadly to enhance user engagement and conversion rates.

- **Average Spending:** The analysis did not reveal a significant impact on average spending. While the treatment improved conversion rates, it did not affect how much users spend on average. Further investigation into spending behavior or alternative strategies may be necessary to influence user spending.

**Conclusion:**

The A/B test results demonstrate a significant positive impact of the treatment on conversion rates, indicating its effectiveness in driving user actions. However, the lack of significant changes in average spending suggests that additional measures or different approaches might be required to influence spending behavior. Therefore, while the treatment should be adopted for improved conversion rates, further efforts should focus on understanding and enhancing factors that affect spending.
 and enhancing factors that affect spending.
 required to influence spending behavior.